lib/trino-filesystem/src/main/java/io/trino/filesystem/encryption/EncryptionKey.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.encryption;
+
+import java.util.Arrays;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static java.util.Objects.requireNonNull;
+
+public record EncryptionKey(byte[] key, String algorithm)
+{
+    public EncryptionKey {
+        requireNonNull(algorithm, "algorithm is null");
+        key = Arrays.copyOf(key, key.length);
```

Comment:
```java
key = key.clone()

Reply:
Actually... let's just let it be mutable and only check for null here

Otherwise we have to copy it every time it is accessed